mhw_analysis/ai/ai_mhw_images.py

```diff
@@ -0,0 +1,295 @@
+# create module paths
+MHW_PATH = '/Users/kamil/Desktop/research2020/mhw_analysis'
+OCEANPY_PATH = '/Users/kamil/Desktop/research2020/oceanpy'
+os.environ['NOAA_OI'] = '/Users/kamil/Desktop/NOAA-OI-SST-V2'
+
+# append paths
+sys.path.append(MHW_PATH)
+sys.path.append(OCEANPY_PATH)
+
+from mhw_analysis.cems import io as cems_io
+from oceanpy.sst import io as sst_io
+
+from datetime import timedelta
+# from datetime import date
+from datetime import datetime
+from pandas import DataFrame
+
+import pandas
+import sqlalchemy
+from sqlalchemy.orm import sessionmaker
+
+# import seaborn	
+import pickle
+
+def select_range(connection, table, lat_start=30.0, lat_end=35.0, lon_start=30.0, lon_end=35.0, date_start='Jun 1 2005', date_end='Jun 1 2006'):
+  """
+  Queries MHW_Events table given ranges (latitude, longitude, date), and returns
+  a dataframe containing all the entries of the table which meet that criteria. 
+
+  Query (SQL):
+  SELECT * FROM MHW_Events 
+  WHERE lat BETWEEN ? AND ? 
+  AND lon BETWEEN ? AND ?
+  AND date BETWEEN ? AND ?
+
+  :param table:
+    SQLAlchemy Table object containing MHW_Events
+    Run the following code before calling this function:
+      engine = sqlalchemy.create_engine('sqlite:///'+mhw_file)
+      metadata = sqlalchemy.MetaData()
+      mhw_tbl = sqlalchemy.Table('MHW_Events', metadata, autoload=True, autoload_with=engine)
+  
+  :param lat_start:
+    double representing the lower end of latitude range
+  
+  :param lat_end:
+    double representing the upper end of latitude range
+  
+  :param lon_start:
+    double representing the lower end of longitude range
+  
+  :param lon_end:
+    double representing the lower end of longitude range
+  
+  :param date_start:
+    String representing the start date
+    Format: 'Mon Day Year' where Day and Year are integers
+  
+  :param date_end:
+    String representing the start date
+    Format: 'Mon Day Year' where Day and Year are integers
+  
+  :return range_df:
+    Pandas dataframe containing query results
+  """
+
+  # can't run if table not properly set up
+  # if not table:
+  #   print("Please load the MHW_Events table into a Table object.")
+  #   return None
+  
+  # set up query
+  query = sqlalchemy.select([table]).where(sqlalchemy.and_(
+    # range of lat 
+    sqlalchemy.between(
+        # pass functions?
+        table.columns.lat, lat_start, lat_end
+    ),
+    # and range of lon
+    sqlalchemy.between(
+        # pass functions?
+        table.columns.lon, lon_start, lon_end
+    ),
+    # and range of dates
+    sqlalchemy.between(
+        # pass functions?
+        table.columns.date, datetime.strptime(date_start, '%b %d %Y'), datetime.strptime(date_end, '%b %d %Y')
+    )
+    ))
+  
+  # execute query
+  result = connection.execute(query)
+
+  # get df
+  range_df = DataFrame(result.fetchall())
+  # range_df.columns = query.keys()
+
+  # return df
+  return range_df
+
+# used to save the tuples 
+# could include folder path as a param
```

Comment:
Yes, please have a folder path.  

Best not to save anything into the Repo if we can avoid it.